### PR TITLE
Added support for @require-extends phpdoc

### PR DESCRIPTION
**Pull request by staabm**

### CODE REVIEW

**Comment by staabm** [comment]
should I do `@require-implements` in this very same PR?

edit: just realized I am missing the "printing" parts

File: src/Ast/PhpDoc/PhpDocNode.php [hunk 1]
```diff
@@ -187,6 +187,19 @@ static function (PhpDocTagValueNode $value): bool {
 		);
 	}
 
+	/**
+	 * @return RequireExtendsTagValueNode[]
+	 */
+	public function getRequireExtendsTagValues(string $tagName = '@require-extends'): array
```
Comment (ondrejmirtes) [inline src/Ast/PhpDoc/PhpDocNode.php]: Similar tag has a prefix here by default: https://github.com/phpstan/phpdoc-parser/blob/a3b405864280bbae461c89e2e6a47ffee3fb86c9/src/Ast/PhpDoc/PhpDocNode.php#L291 (We don't need to support and test prefix-less variant).

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
Yes, please add `@phpstan-require-implements` here too.

**Review by ondrejmirtes — APPROVED** [review]

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**